SalesforceCommerceCloud/pwa-kit
# W-15586246 Remove out of stock products on cart

# Description
Some products can be out of stock during shopping journey. We want to make sure out of stock will also removed before allowing shoppers to navigate from Cart to checkout page. At the moment, we have the unavailableModal which will promot shoppers to remove unavailable products, we should include out of stock products

Related issue: https://github.com/SalesforceCommerceCloud/pwa-kit/issues/1759



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Check out code
- Run Retail App
- Add an item to cart
- Go to BM, change the item availability to be Unavailable
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/476a239e-bc47-4ee9-9ed0-a1489f751614)
- Go back to Retail app Cart Page
- Out of stock items should be prompted for removal

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
